src/Microsoft.TemplateEngine.Cli/TemplateListResolver.cs

```diff
@@ -321,23 +260,20 @@ private static IReadOnlyList<IFilteredTemplateInfo> FilterTemplatesOnParameters(
                         {   // the name is a known template param, it must have not parsed due to an invalid value
                             //
                             // Note (scp 2017-02-27): This probably can't happen, the param parsing doesn't check the choice values.
-                            dispositionForTemplate.Add(new MatchInfo { Location = MatchLocation.OtherParameter, Kind = MatchKind.InvalidParameterValue, ChoiceIfLocationIsOtherChoice = unmatchedParamName });
+                            template.AddDisposition(new MatchInfo { Location = MatchLocation.OtherParameter, Kind = MatchKind.InvalidParameterValue, ChoiceIfLocationIsOtherChoice = unmatchedParamName });
                         }
                         else
                         {   // the name is not known
-                            dispositionForTemplate.Add(new MatchInfo { Location = MatchLocation.OtherParameter, Kind = MatchKind.InvalidParameterName, ChoiceIfLocationIsOtherChoice = unmatchedParamName });
+                            // TODO: reconsider storing the canonical in this situation. It's not really a canonical since the param is unknown.
+                            template.AddDisposition(new MatchInfo { Location = MatchLocation.OtherParameter, Kind = MatchKind.InvalidParameterName, ChoiceIfLocationIsOtherChoice = unmatchedParamName });
                         }
                     }
                 }
-                catch
+                catch (CommandParserException ex)
```

Comment:
Are there any other cases that could throw here? Should we add a general catch in addition to the specific one that has something to the effect of "unexpected error"? We could end up dumping a stack here where we wouldn't have before

Reply:
Adding a general catch, but it's going to do the same thing as CommandParserException for now. Not much else we can do  in this situation. We could create another MatchKind and/or MatchLocaltion in addition to "Unspecified", have the caller of this check for that sort of disposition, and bail with an appropriate message. Not sure what's best, at this point, a template that catches and adds the unspecified disposition merely gets treated as a non-match.